[docs][yugabyted][2024.1.2] Add Read Replica and xCluster examples (#23289)
yugabyte/yugabyte-db: docs/content/preview/reference/configuration/yugabyted.md

Use the `yugabyted start` command to start a one-node YugabyteDB cluster for running [YSQL](../../../api/ysql) and [YCQL](../../../api/ycql) workloads in your local environment.

To use encryption in transit, OpenSSL must be installed on the nodes.

> If you want to use backup and restore, start the node with `--backup_daemon=true` to initialize the backup and restore agent. You also need to download and extract the [YB Controller release](https://s3.us-west-2.amazonaws.com/downloads.yugabyte.com/ybc/2.1.0.0-b9/ybc-2.1.0.0-b9-linux-x86_64.tar.gz) to the yugabyte-{{< yb-version version="preview" >}} folder.

Reviewer: can we please update the download string to the following - 

```
https://downloads.yugabyte.com/ybc/2.1.0.0-b9/ybc-2.1.0.0-b9-linux-x86_64.tar.gz
```

Reviewer: Change from  - 

```
to the yugabyte-{{< yb-version version="preview" >}} folder.
```
to -

```
to the yugabyte-{{< yb-version version="preview" >}} release directory.

: Use this flag to start a read replica node.

--backup_daemon *backup-daemon-process*
: Enable or disable the backup daemon with yugabyted start. Default: `false`

> : If you start a cluster using the `--backup_daemon` flag, you also need to download and extract the [YB Controller release](https://s3.us-west-2.amazonaws.com/downloads.yugabyte.com/ybc/2.1.0.0-b9/ybc-2.1.0.0-b9-linux-x86_64.tar.gz) to the yugabyte-{{< yb-version version="preview" >}} folder.

Reviewer: Can we please update the download string to the following -

```
https://downloads.yugabyte.com/ybc/2.1.0.0-b9/ybc-2.1.0.0-b9-linux-x86_64.tar.gz
```

--log_dir *log-directory*
: The log directory for the yugabyted server whose version is desired.

### xcluster

Use the `yugabyted xcluster` command to set-up, or delete a [xcluster](../../../architecture/docdb-replication/async-replication/) replication.

#### Syntax

```text
Usage: yugabyted xcluster [command] [flags]
```

#### Commands

The following sub-commands are available for the `yugabyted xcluster` command:

- [checkpoint](#checkpoint)
- [set_up](#set-up)
- [status](#status-1)
- [delete](#delete-1)

#### checkpoint

> Use the sub-command `yugabyted xcluster checkpoint` to checkpoint a new xcluster replication between 2 clusters. This command needs to be run from the source cluster of the replication.

Reviewer: This is a new xcluster feature in the coredb. Checkpoint will take a snapshot of the DB and determine whether bootstrapping is required for the remote cluster.

For example, to create a new xcluster replication, execute the following command:

```sh
./bin/yugabyted xcluster checkpoint --replication_id <replication_id> --databases <comma_seperated_database_names>
```

##### checkpoint flags

-h | --help
: Print the command-line help and exit.

--base_dir *base-directory*
: The base directory for the yugabyted server.

--databases *xcluster-databases*
: Comma separated list of databases to be added to the replication.

--replication_id *xcluster-replication-id*

> : A unique string to be defined by you which will be used to uniquely identify the replication.

Reviewer 2: User creates the replication ID using this command? Or enters an existing ID?

To create a multi-zone cluster:

> 1. Start the first node by running the `yugabyted start` command, passing in the `--cloud_location` and `--fault_tolerance` flags to set the node location details. Please use `--backup_daemon` flag if backup/restore operations (maybe required for xCluster replication) will be performed on this cluster. No value needs to be passed through `--backup_daemon` flag:

Reviewer 2: ```suggestion
1. Start the first node by running the `yugabyted start` command, passing in the `--cloud_location` and `--fault_tolerance` flags to set the node location details.

     Add the `--backup_daemon` flag if you want to perform backup and restore operations (also required for xCluster replication) on this cluster. The `--backup_daemon` flag doesn't take a value.
```

    ```sh
    ./bin/yugabyted start --advertise_address=<host-ip> \
        --cloud_location=aws.us-east-1.us-east-1a \
        --fault_tolerance=zone
    ```

> 1. Start the second and the third node on two separate VMs using the `--join` flag. Please use `--backup_daemon` flag if backup/restore operations (maybe required for xCluster replication) will be performed on this cluster. No value needs to be passed through `--backup_daemon` flag:

Reviewer 2: ```suggestion
1. Start the second and the third node on two separate VMs using the `--join` flag.

    Add the `--backup_daemon` flag if you want to perform backup and restore operations (also required for xCluster replication) on this cluster. The `--backup_daemon` flag doesn't take a value. 
```

To create a secure multi-region cluster:

> 1. Start the first node by running the `yugabyted start` command, using the `--secure` flag and passing in the `--cloud_location` and `--fault_tolerance` flags to set the node location details. Please use `--backup_daemon` flag if backup/restore operations (maybe required for xCluster replication) will be performed on this cluster. No value needs to be passed through `--backup_daemon` flag:

Reviewer 2: ```suggestion
1. Start the first node by running the `yugabyted start` command, using the `--secure` flag and passing in the `--cloud_location` and `--fault_tolerance` flags to set the node location details.

    Add the `--backup_daemon` flag if you want to perform backup and restore operations (also required for xCluster replication) on this cluster. The `--backup_daemon` flag doesn't take a value. 
```

    - Copy the certificates for the second VM from `$HOME/var/generated_certs/<IP_of_VM_2>` in the first VM to `$HOME/var/certs` in the second VM.
    - Copy the certificates for third VM from `$HOME/var/generated_certs/<IP_of_VM_3>` in first VM to `$HOME/var/certs` in the third VM.

> 1. Start the second and the third node on two separate VMs using the `--join` flag. Please use `--backup_daemon` flag if backup/restore operations (maybe required for xCluster replication) will be performed on this cluster. No value needs to be passed through `--backup_daemon` flag:

Reviewer 2: ```suggestion
1. Start the second and the third node on two separate VMs using the `--join` flag.

    Add the `--backup_daemon` flag if you want to perform backup and restore operations (also required for xCluster replication) on this cluster. The `--backup_daemon` flag doesn't take a value. 
```

To create a multi-region cluster:

> 1. Start the first node by running the `yugabyted start` command, pass in the `--cloud_location` and `--fault_tolerance` flags to set the node location details. Please use `--backup_daemon` flag if backup/restore operations (maybe required for xCluster replication) will be performed on this cluster. No value needs to be passed through `--backup_daemon` flag:

Reviewer 2: ```suggestion
1. Start the first node by running the `yugabyted start` command, pass in the `--cloud_location` and `--fault_tolerance` flags to set the node location details.

    Add the `--backup_daemon` flag if you want to perform backup and restore operations (also required for xCluster replication) on this cluster. The `--backup_daemon` flag doesn't take a value.
```

    ```sh
    ./bin/yugabyted start --advertise_address=<host-ip> \
        --cloud_location=aws.us-east-1.us-east-1a \
        --fault_tolerance=region
    ```

> 1. Start the second and the third node on two separate VMs using the `--join` flag. Please use `--backup_daemon` flag if backup/restore operations (maybe required for xCluster replication) will be performed on this cluster. No value needs to be passed through `--backup_daemon` flag:

Reviewer 2: ```suggestion
1. Start the second and the third node on two separate VMs using the `--join` flag.

    Add the `--backup_daemon` flag if you want to perform backup and restore operations (also required for xCluster replication) on this cluster. The `--backup_daemon` flag doesn't take a value. 
```
